docs/state.md

```diff
@@ -40,25 +40,29 @@ The worker maintains a worker state which is a JSON representation of the follow
 
 ```jsonc
 {
-    "worker_id": "<WORKER_ID>"
+    "worker_id": "<WORKER_ID>",
+    "instance_id": "<INSTANCE_ID>"  # OPTIONAL
 }
 ```
 
 #### `worker_id`
 
-This contains a unique identifier that represents the worker in your AWS Deadline Cloud fleet.
+This contains a unique identifier that represents the worker in your AWS Deadline Cloud fleet, as 
+well as an optional identifier of the EC2 instance running the worker, if applicable. 
 
-When the agent starts up, it checks for the worker state file and this value. If a `worker_id` is
-found, the worker agent will transition the worker to the `RUNNING` state and begin running as the
+When the agent starts up, it checks for the worker state file and these values. If a `worker_id` is 
+found, and an `instance_id` in the file either matches that of the running host or is not present, 
+the worker agent will transition the worker to the `RUNNING` state and begin running as the
 worker.
 
-If the worker state file is not found or contains no value for `worker_id`, then the worker agent
-will create a new worker in the AWS Deadline Cloud fleet and persist the `worker_id` to the worker
-state file.
+If the worker state file is not found, contains no value for `worker_id`, or contains a `worker_id` 
+with an `instance_id` that is different than the `instance_id` detected on the running host, then 
+the worker agent will create a new worker in the AWS Deadline Cloud fleet and persist the 
+`worker_id` and `instance_id` to the worker state file.
```

Comment:
I really like the state machine diagram you have in the PR description. Can we copy it in here to help provide a visual aid to readers?